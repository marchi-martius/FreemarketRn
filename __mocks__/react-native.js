const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

module.exports = rn;
