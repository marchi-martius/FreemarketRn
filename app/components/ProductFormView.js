import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Button, StyleSheet, Keyboard } from 'react-native';
import t from 'tcomb-form-native';

import FormImagePicker from './FormImagePicker';

const styles = StyleSheet.create({
  formView: {
    padding: 10,
  },
  button: {
    paddingTop: 10,
    paddingBottom: 10,
  },
});

const Form = t.form.Form;

const Product = t.struct({
  name: t.String,
  price: t.Number,
});

export default class ProductFormView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageSrouce: null,
      localImage: null,
    };
  }

  onFormPress() {
    const product = this.form.getValue();
    const { localImage } = this.state;

    Keyboard.dismiss();
    if (product) {
      this.props.onPress({ product, localImage });
    }
  }

  setLocalImage(localImage) {
    this.setState({
      imageSource: localImage.uri,
      localImage,
    });
  }

  render() {
    return (
      <View style={styles.formView}>
        <FormImagePicker
          imageSource={this.state.imageSource}
          setLocalImage={localImage => this.setLocalImage(localImage)}
        />
        <Form
          ref={_form => (this.form = _form)}
          type={Product}
        />
        <View style={styles.button}>
          <Button
            onPress={() => this.onFormPress()}
            title="Add Product"
          />
        </View>
        <View style={styles.button}>
          <Button
            style={styles.button}
            onPress={() => this.props.onBackPress()}
            color="red"
            title="Back"
          />
        </View>
      </View>
    );
  }
}

ProductFormView.propTypes = {
  onPress: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
};
