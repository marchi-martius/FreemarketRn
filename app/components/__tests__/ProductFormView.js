import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import ProductFormView from '../ProductFormView';

it('renders successfully', () => {
  const tree = renderer.create(
    <ProductFormView
      onPress={() => true}
      onBackPress={() => true}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
