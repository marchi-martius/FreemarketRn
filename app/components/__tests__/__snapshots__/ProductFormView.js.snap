// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 1,
            "borderStyle": "dotted",
            "borderWidth": 4,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Place for an Image
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
          "width": 150,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Select Image
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Name
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Name"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Price
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Price"
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType="numeric"
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Product
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</View>
`;
